tests/tools/agent_runner.py

```diff
@@ -0,0 +1,144 @@
+from typing import Dict, Optional, Mapping, Callable
+from pathlib import Path
+import abc
+import json
+import tempfile
+
+from .agent_file import MountedFile
+from .utils import create_temp_dir
+
+
+class BaseContainerRunner(abc.ABC):
+    def __init__(self, name, image, docker_client, command=None):
```

Comment:
A comment / docstring which describes on a high level what that class does would be nice.